recipes/hiredis/0.14.1/conanfile.py

```diff
@@ -0,0 +1,78 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+
+from os import rename
+from os.path import join
+
+
+class HiredisConan(ConanFile):
+    name = "hiredis"
+
+    description = "Minimalistic C client for Redis >= 1.2"
+    topics = "conan", "c", "redis"
+
+    license = "BSD 3-Clause \"New\" or \"Revised\" License"
+    settings = "os", "arch", "compiler", "build_type"
+
+    homepage = "https://github.com/redis/hiredis"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    options = {
+        "fPIC": [True, False],
+        "shared": [True, False]
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": True
+    }
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = self.name + "-" + self.version
+        rename(extracted_folder, "src")
+
+    def build(self):
+        # settings
+        isWindows = self.settings.os == "Windows"
+
+        # update makefile
+        makefile = join(self.source_folder, "src", "Makefile")
+        tools.replace_in_file(makefile, "-fPIC ", "", strict=False)
+
+        with tools.chdir(join(self.source_folder, "src")):
+            autoTools = AutoToolsBuildEnvironment(self, win_bash=isWindows)
+            autoTools.make()
+
+    def package(self):
+        # settings
+        isWindows = self.settings.os == "Windows"
+
+        with tools.chdir(join(self.source_folder, "src")):
+            autoTools = AutoToolsBuildEnvironment(self, win_bash=isWindows)
```

Comment:
Please move the creation (and eventual configuration) of `AutoToolsBuildEnvironment` to a `_configure_autotools` helper method.
This avoid code copying for its arguments.